File: src/main/java/com/lordstdpa/domain/booking/BookingRepository.java
package com.lordstdpa.domain.booking;

import com.lordstdpa.data.booking.BookingDataModel;

import java.time.LocalDateTime;

public interface BookingRepository {
    void createBooking(BookingDataModel bookingDataModel);

    void findByBookingId(String bookingId);

    BookingDataModel findBy(String numberPlate, LocalDateTime start);
}

File: travel_book_flutter_client/lib/data/booking/booking_repository.dart
import 'package:travel_book_flutter_client/data/booking/booking.dart';
import 'package:travel_book_flutter_client/data/booking/booking_service.dart';

class BookingRepository {
  final BookingService _bookingService;

  BookingRepository({required BookingService bookingService})
      : _bookingService = bookingService;

  Future<List<Booking>> findBookingsInRange(
      {required DateTime start, DateTime? end}) async {
    return _bookingService.findBy(start: start, end: end);
  }
}
